bundles/org.openhab.binding.samsungtv/src/main/java/org/openhab/binding/samsungtv/internal/service/SmartThingsApiService.java

```diff
@@ -0,0 +1,981 @@
+                throw new IOException("No Data - is DeviceID correct?");
+            }
+            tvValues.filter(t -> t.isError()).ifPresent(t -> logger.debug("{}: Error: {}", host, t.getError()));
+            errorCount = 0;
+        } catch (JsonSyntaxException | URISyntaxException | IOException e) {
+            logger.debug("{}: Cannot connect to Smartthings Cloud: {}", host, e.getMessage());
+            if (errorCount++ > MAX_ERRORS) {
+                logger.warn("{}: Too many connection errors, disabling SmartThings", host);
+                stop();
+            }
+        }
+        return tvValues;
+    }
+
+    /**
+     * Smartthings API HTTP setter
+     * Currently rate limited to 350 requests/minute
+     *
+     * @param capability eg mediaInputSource
+     *
+     * @param command eg setInputSource
+     *
+     * @param value from acceptible list eg HDMI1, digitalTv, AM etc
+     *
+     * @return boolean true if successful
+     */
+    public synchronized boolean setTVProperties(String capability, String command, String value) {
+        if (apiKey.isBlank() || deviceId.isBlank()) {
+            return false;
+        }
+        Optional<String> response = Optional.empty();
+        try {
+            String contentString = new Gson().toJson(new JSONContent(capability, command, value));
+            logger.trace("{}: content: {}", host, contentString);
+            InputStream content = new ByteArrayInputStream(contentString.getBytes());
+            String api = API_ENDPOINT_V1 + "devices/" + deviceId + COMMAND;
+            URI uri = new URI("https", null, SMARTTHINGS_URL, 443, api, null, null);
+            response = sendUrl("POST", uri, content);
+        } catch (JsonSyntaxException | URISyntaxException | IOException e) {
+            logger.debug("{}: Send Command to Smartthings Cloud failed: {}", host, e.getMessage());
+        }
+        return response.map(r -> r.contains("ACCEPTED") || r.contains("COMPLETED")).orElse(false);
+    }
+
+    /**
+     * Smartthings API Subscription
+     *
+     *
+     * @return Optional<STSubscription> stSub
+     */
+    public synchronized Optional<STSubscription> smartthingsSubscription() {
+        if (apiKey.isBlank() || deviceId.isBlank()) {
+            return Optional.empty();
+        }
+        Optional<STSubscription> stSub = Optional.empty();
+        try {
+            logger.info("{}: SSE Creating Smartthings Subscription", host);
+            String contentString = new Gson().toJson(new JSONSubscriptionFilter(deviceId));
+            logger.trace("{}: subscription: {}", host, contentString);
+            InputStream subscriptionFilter = new ByteArrayInputStream(contentString.getBytes());
+            URI uri = new URI("https", null, SMARTTHINGS_URL, 443, "/subscriptions", null, null);
+            Optional<String> response = sendUrl("POST", uri, subscriptionFilter);
+            stSub = response.map(r -> new Gson().fromJson(r, STSubscription.class));
```

Comment:
Isn't it needed to do a null/empty check before creating the STSubscription from the json?

Reply:
No, because it's an Optional. It's present or it's empty, it can't be null.

Reply:
I know, but i woudl have expected an empty string to throw a `JsonSyntaxException`. Now i have read the documentation and if i understand it correctly the desrializer is not called for an empty json string.